types/src/get_with_proof.rs

```diff
@@ -160,27 +158,29 @@ impl<Sig: Signature> UpdateToLatestLedgerResponse<Sig> {
     /// verification.
     pub fn verify(
         &self,
-        validator_verifier: Arc<ValidatorVerifier<Sig::VerifyingKeyMaterial>>,
+        epoch_info: &mut EpochInfo,
         request: &UpdateToLatestLedgerRequest,
     ) -> Result<()> {
         verify_update_to_latest_ledger_response(
-            validator_verifier,
+            epoch_info,
             request.client_known_version,
             &request.requested_items,
             &self.response_items,
             &self.ledger_info_with_sigs,
+            &self.validator_change_events,
         )
     }
 }
 
 /// Verifies content of an [`UpdateToLatestLedgerResponse`] against the proofs it
 /// carries and the content of the corresponding [`UpdateToLatestLedgerRequest`]
-pub fn verify_update_to_latest_ledger_response<Sig: Signature>(
-    validator_verifier: Arc<ValidatorVerifier<Sig::VerifyingKeyMaterial>>,
+pub fn verify_update_to_latest_ledger_response(
+    epoch_info: &mut EpochInfo,
```

Comment:
Would be it be cleaner to return a new epoch_info instead of in-place mutation of a parameter?

Reply:
don't have strong preference, I'll update